src/nat/llm/huggingface_llm.py

```
@@ -0,0 +1,286 @@
+                del cached["model"]
+            
+            # Clear CUDA cache if available
+            if "torch" in cached and hasattr(cached["torch"].cuda, "empty_cache"):
+                cached["torch"].cuda.empty_cache()
+            
+            # Remove from cache
+            del _model_cache[model_name]
+            
+            logger.debug("Model cleaned up: %s", model_name)
+    except Exception as e:
+        logger.exception("Error cleaning up HuggingFace model '%s': %s", model_name, e)
```

Ruff (0.14.8)

Redundant exception object included in `logging.exception` call

(TRY401)